### PR TITLE
Adding ignore paths to cache test

### PR DESCRIPTION
**Pull request by ricardoamador**
Adding exclusion paths to the Mac builder_cache target to avoid archiving symlinks in the SDKs or other unwanted packages. Symlinks cause issues with the CAS archiving functionality. We have opened a bug for the LUCI team regarding this as well.

Companion CL in recipes is here: https://flutter-review.googlesource.com/c/recipes/+/54110

*List which issues are fixed by this PR. You must list at least one issue.*
Fixes https://github.com/flutter/flutter/issues/141688

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or the PR is [test-exempt]. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: .ci.yaml
Comment (keyonghan) [inline .ci.yaml:97]: Should this be under `Mac builder_cache`?

Comment (ricardoamador) [reply .ci.yaml:97]: Oh right! Thank you! Good catch.

**Review by keyonghan — APPROVED** [review]
LGTM.

After this lands, let's monitor a bit and then bring the `Mac builder_cache` back.

File: .ci.yaml
Comment (zanderso) [inline .ci.yaml:133]: Reminder to update https://github.com/flutter/cocoon/blob/main/CI_YAML.md#properties. It looks like some of the other properties might also be missing docs there?

**Review by zanderso — APPROVED** [review]

File: .ci.yaml
Comment (ricardoamador) [reply .ci.yaml:133]: You got it! I will make sure to open that before submitting.

**Comment by ricardoamador** [comment]
Ran a few tests but could not get the condition of the cache needing an update. So will merge then switch back to prod after some monitoring. Leaving in staging for now is safer.

**Pull request merged**
